### PR TITLE
Update Node.js requirement in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by NotMyFault**
We're using Node 18.x to build core.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7550"></a>

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

File: CONTRIBUTING.md
Comment (timja) [inline CONTRIBUTING.md:18]: it is needed for the `Building frontend assets` section though I think. the maven frontend node / yarn won't be in the users path.

Comment (NotMyFault) [reply CONTRIBUTING.md]: I see, thought people would build Jenkins first, before doing anything else particularly.

Quickly changed the purpose of this PR then :P

File: CONTRIBUTING.md
Comment (StefanSpieker) [inline CONTRIBUTING.md:18]: Accoring to the [docs](https://yarnpkg.com/getting-started/install#install-corepack) it should be installed with corepack:

```suggestion
     - Frontend tasks are run using [yarn](https://yarnpkg.com/). Run `corepack enable` to install corepack as the recommended way and afterwards `corepack prepare yarn@stable --activate` to install yarn.
```

Comment (timja) [reply CONTRIBUTING.md:18]: `corepack enable` from the war directory should be all that's needed as the version is defined in the package.json https://github.com/jenkinsci/jenkins/blob/f0952735b523ff5af6aab3f0a7b214964b5a16a9/war/package.json#L68

Comment (StefanSpieker) [reply CONTRIBUTING.md:18]: Oh, that is nice. I was not aware of that -  tested it and it works fine

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
